### PR TITLE
Add issue templates for Pyodide Github repository

### PR DESCRIPTION
**Pull request by ryanking13**
This PR adds issue templates for the Pyodide Github repository.
Issues that applied these templates can be seen [here](https://github.com/ryanking13/pyodide/issues/new/choose).

Most templates are borrowed from [Pytorch-Lightning](https://github.com/PyTorchLightning/pytorch-lightning/tree/master/.github/ISSUE_TEMPLATE).

- Bug report
- Documentation
- Feature Request
- Package Request
- Code Improvement
- (+) Link to discussion
- (+) Link to gitter

Closes: #1844
CC: @sylvain-ri

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md
Comment (rth) [inline .github/ISSUE_TEMPLATE/bug_report.md:25]: I think examples should be in comments, otherwise people will leave them in and it will add noise to the bug report,

```suggestion
- Pyodide version: <!-- e.g. 1.8.1 -->
- Browser version: <!-- e.g. Chrome 95.0.4638.54 -->
- Any other relevant information:
```

File: .github/ISSUE_TEMPLATE/bug_report.md
Comment (rth) [inline .github/ISSUE_TEMPLATE/bug_report.md:30]: Same here.

File: .github/ISSUE_TEMPLATE/config.yml
Comment (rth) [inline .github/ISSUE_TEMPLATE/config.yml:3]: I think for usage questions we should also recommend StackOverflow https://stackoverflow.com/questions/tagged/pyodide it's a bit better indexed by search engines and it's good to have more people answering aside from contributors.

The way I see it GH Discussions is more for general discussions  and StackOverflow is more more specific questions: "how to I do ... with Pyodide".

File: .github/ISSUE_TEMPLATE/documentation.md
Comment (rth) [inline .github/ISSUE_TEMPLATE/documentation.md:16]: I think in most cases it's fine if people submit the PR directly and check the rendering in CI. 

Then for larger PRs mention that they can indeed build the docs locally if needed.

File: .github/ISSUE_TEMPLATE/documentation.md
Comment (rth) [inline .github/ISSUE_TEMPLATE/documentation.md:13]: I think all this is should be escaped as a comment maybe?

File: .github/ISSUE_TEMPLATE/documentation.md
Comment (rth) [inline .github/ISSUE_TEMPLATE/documentation.md:18]: maybe,
```suggestion
<!-- Note: If there are no changes in code, we recommend you to append `[skip ci]` in your commit message to avoid starting the build CI.
-->
```

**Review by rth — COMMENTED** [review]
Thanks a lot for proposing this @ryanking13 ! A few comments below.

I also think that anything that is a comment explaining what people should do (or examples of responses), need to be escaped with `<!-- -->`. Otherwise you can assume that it will be left in the issue when submitted and we will end up with lots of boilerplate text when processing issues.

File: .github/ISSUE_TEMPLATE/documentation.md
Comment (hoodmane) [reply .github/ISSUE_TEMPLATE/documentation.md:18]: Though it would be really great if we could figure out how to unconditionally run format checks. I guess last time I said this the status was "not clear that it is possible".

Comment (hoodmane) [reply .github/ISSUE_TEMPLATE/documentation.md:18]: Maybe one approach would be we could add the a manually triggered format checker to github actions and then run that on all `[skip ci]` PRs before merging?

Comment (rth) [reply .github/ISSUE_TEMPLATE/documentation.md:18]: Hmm, well we can always go back to custom code that parses the commit message and kills CI it if required. That should be even easier in GH actions. Since manually triggered CI is also not ideal.

Comment (hoodmane) [reply .github/ISSUE_TEMPLATE/documentation.md:18]: Well we'd need it for both GH actions and CircleCI if we have both. For CircleCI, I guess the easy thing would be to make a new ancestor of all of the checks you don't want to run and fail it if the run isn't needed. But it would be nice to have "don't run dependent jobs but report as success".

Surely it's common for monorepos to have many independent subsystems and for people to only want to test the subsystem that changed? 

Anyways this discussion is a bit off topic here.

File: .github/ISSUE_TEMPLATE/bug_report.md
Comment (phorward) [reply .github/ISSUE_TEMPLATE/bug_report.md:30]: Yep.

```suggestion
- Build environment: <!-- e.g. Ubuntu 18.04, Windows 10, pyodide/pyodide-env:19 docker -->
```

**Comment by ryanking13** [comment]
Applied suggestions

File: .github/ISSUE_TEMPLATE/package_request.md
Comment (rth) [inline .github/ISSUE_TEMPLATE/package_request.md:14]:
```suggestion
- Package Name and Version <!-- (e.g. pandas / latest) -->:
- Package URL <!--  (e.g. github link, PyPI link) -->:
```

File: .github/ISSUE_TEMPLATE/documentation.md
Comment (rth) [inline .github/ISSUE_TEMPLATE/documentation.md:3]:
```suggestion
name: Documentation improvement
about: Improve or fix the documentation
```

**Review by rth — APPROVED** [review]
A few more minor comments otherwise LGTM.

I'll merge it in a day or so, unless there are other review comments.

**Pull request merged**
